src/qibolab/compilers/compiler.py

```diff
@@ -129,12 +129,19 @@ def compile(self, circuit, platform):
 
         measurement_map = {}
         # process circuit gates
+        delays = defaultdict(int)
```

Comment:
Is it meaningful to preserve the delay across different moments?

Reply:
I also had the impression that we can avoid the dictionary, but I think it is needed because the circuit will look like
```
--G1-------
--Al--G3--
--G2-------
```
so the `delay` associated with `Al` needs to be introduced in `G3` which is in the next moment.

Reply:
More than avoiding, I was thinking to move inside one more loop (i.e. the first).

But I guess you're right, it's impossible.